R/segmentcomparisontable.R

```diff
@@ -0,0 +1,289 @@
+        if (v.qtype %in% c("PickOne", "Date"))
+            tmp <- sweep(tmp, 2, colSums(tmp), "/")
+
+        # Use question attributes to determine row names
+        if (v.qtype == "Date")
+            rownames(tmp) <- levels(attr(x[[vvi]], "QDate"))
+        else if (v.qtype == "PickOne")
+            rownames(tmp) <- levels(x[[vvi]])
+        else if (v.qtype %in% c("PickAny", "NumberMulti") && ncol(vv) > 1)
+            rownames(tmp) <- tmp.colnames
+        else
+            rownames(tmp) <- attr(x[[vvi]], "label")
+
+        if (NROW(tmp) == 1 && rownames(tmp)[1] == attr(x[[vvi]], "question"))
+            rownames(tmp) <- ""
+
+
+        tmp.nvar <- ncol(vv)
+        row.vcol <- c(row.vcol, 1:tmp.nvar)
+        row.span[[length(row.span) + 1]] <- list(label = attr(x[[vvi]], "question"), height = tmp.nvar)
+        if (!is.null(rownames(tmp)))
+            row.labels <- c(row.labels, rownames(tmp))
+        tmp.numeric <- isTRUE(attr(x[[vvi]], "questiontype") %in% c("NumberMulti", "Number"))
+        row.format <- c(row.format, rep(if (tmp.numeric) "numeric" else "percentage", tmp.nvar))
+        row.vvi <- c(row.vvi, rep(vvi, tmp.nvar))
+        v.list[[vvi]] <- vv # save subsetted variable
+
+        # Always compute index values because they are used for cell fill
+        if (!tmp.numeric)
+        {
+            ind.not.missing <- !is.na(group)
+            tot.mean <- Mean(vv[ind.not.missing,,drop = FALSE], weights = weights[ind.not.missing])
+            if (NROW(tmp) > 1)
+                index.values[[vvi]] <- sweep(tmp, 1, tot.mean, "/")
+            else
+                index.values[[vvi]] <- tmp/tot.mean
+
+            if (show.index.values)
+                tmp <- index.values[[vvi]]
+        }
+        result <- rbind(result, tmp)
+    }
+
+    rownames(result) <- NULL # result is preserved in numeric form for exporting
+    result.formatted <- matrix("", nrow(result), ncol(result))
+    for (i in 1:nrow(result))
+        result.formatted[i,] <- if (row.format[i] == "numeric") formatC(result[i,], if (i == 1) 0 else format.numeric.decimals, format = "f", big.mark = ",")
+                                else                            paste0(formatC(result[i,] * 100, format.percentage.decimals, format = "f", big.mark = ","), "%")
+    result.formatted[!is.finite(result)] <- ""
+    result.formatted <- formatC(result.formatted, format = "s", width = max(nchar(result.formatted)))
+    result.formatted <- gsub(" ", "&nbsp;", result.formatted)
+    colnames(result.formatted) <- colnames(result)
+
+    # Font color is determined by p-values
+    results.font.color = font.color
+    if (font.color.set.if.nonsignificant)
+    {
+        results.font.color <- matrix(font.color, nrow(result), ncol(result))
+        results.font.color[3:nrow(result),] <- font.color.nonsignificant # include p = NA
+        pvals <- matrix(NA, nrow(result), ncol(result))
+        rownames(pvals) <- row.labels
```

Comment:
Code fails here due to a dimension mismatch when PickOneMulti input is used. See Routput throwing error in https://app.displayr.com/Dashboard?project_id=-705348#page=5a67bf81-8b62-4800-8cd1-64675dfba945